plugins/item_tasks/server/rest.py

```diff
@@ -320,3 +323,94 @@ def setSpecFromXml(self, item, xml, setName, setDescription, params):
+            handler='worker_handler', user=self.getCurrentUser())
+
+        jobOptions = {
+            'itemTaskId': folder['_id'],
+            'kwargs': {
+                'task': {
+                    'mode': 'docker',
+                    'docker_image': image,
+                    'container_args': [],
+                    'pull_image': pullImage,
+                    'outputs': [{
+                        'id': '_stdout',
+                        'format': 'text'
+                    }],
+                },
+                'outputs': {
+                    '_stdout': {
+                        'mode': 'http',
+                        'method': 'PUT',
+                        'format': 'text',
+                        'url': '/'.join((getApiUrl(), self.resourceName,
+                                         str(folder['_id']), 'add_json_specs')),
+                        'headers': {'Girder-Token': token['_id']},
+                        'params': {
+                            'image': image,
+                            'pullImage': pullImage
+                        }
+                    }
+                },
+                'jobInfo': utils.jobInfoSpec(job),
+                'validate': False,
+                'auto_convert': False,
+                'cleanup': True
+            }
+        }
+        job.update(jobOptions)
+
+        job = jobModel.save(job)
+        jobModel.scheduleJob(job)
+        return job
+
+    @access.token
+    @autoDescribeRoute(
+        Description('Create item tasks under a folder using a list of JSON specifications.')
+        .modelParam('id', model='folder', force=True)
+        .jsonParam('json', 'The JSON specs.', paramType='body')
+        .param('image', 'The docker image name.', required=True, strip=True)
+        .param('pullImage', 'Whether the image should be pulled from a docker registry. ' +
+               'Set to false to use local images only.',
+               dataType='boolean', required=False, default=True),
+        hide=True
+    )
+    def addJsonSpecs(self, folder, json, image, pullImage, params):
+        self.ensureTokenScopes('item_task.set_task_spec.%s' % folder['_id'])
+        token = self.getCurrentToken()
+        user = self.model('user').load(token['userId'], force=True)
+        for itemTaskSpec in json:
+            logger.info('Configuring item "%s"' % itemTaskSpec['name'])
+            item = self.model('item').createItem(
+                name=itemTaskSpec['name'],
+                creator=user,
+                folder=folder,
+                description=itemTaskSpec.get('description', ''),
+                reuseExisting=True)
+            logger.info(pullImage)
+            logger.info(params)
+            itemTaskSpec['docker_image'] = image
+            itemTaskSpec['pull_image'] = pullImage
+            self.model('item').setMetadata(item, {
+                'itemTaskSpec': itemTaskSpec
```

Comment:
You'll also need to add `'isItemTask': True` in order for this to appear in the list of tasks (this is due to a limitation of mongo indexing where we cannot index for existence only).